src/dotnet-compile/Program.cs

```diff
@@ -316,42 +314,33 @@ private static bool CompileProject(ProjectContext context, CompilerCommandApp ar
             if (success)
             {
                 success &= GenerateCultureResourceAssemblies(context.ProjectFile, dependencies, outputPath);
-            }
+            }            
 
-            bool generateBindingRedirects = false;
-            if (success && !args.NoHostValue && compilationOptions.EmitEntryPoint.GetValueOrDefault())
-            {
-                generateBindingRedirects = true;
-                var rids = PlatformServices.Default.Runtime.GetAllCandidateRuntimeIdentifiers();
-                var runtimeContext = ProjectContext.Create(context.ProjectDirectory, context.TargetFramework, rids);
-                runtimeContext
-                    .MakeCompilationOutputRunnable(outputPath, args.ConfigValue);
-            }
-            else if (!string.IsNullOrEmpty(context.ProjectFile.TestRunner))
+            if (success && ShouldMakeRunnable(context, compilationOptions))
             {
-                generateBindingRedirects = true;
-                var projectContext =
-                    ProjectContext.Create(context.ProjectDirectory, context.TargetFramework,
-                        new[] { PlatformServices.Default.Runtime.GetLegacyRestoreRuntimeIdentifier() });
-
-                // Don't generate a deps file if we're on desktop
-                if (!context.TargetFramework.IsDesktop())
+                var rids = new List<string>();
+                if (string.IsNullOrEmpty(args.RuntimeValue))
                 {
-                    projectContext
-                        .CreateExporter(args.ConfigValue)
-                        .GetDependencies(LibraryType.Package)
-                        .WriteDepsTo(Path.Combine(outputPath, projectContext.ProjectFile.Name + FileNameSuffixes.Deps));
+                    rids.AddRange(PlatformServices.Default.Runtime.GetAllCandidateRuntimeIdentifiers());
+                }
+                else
+                {
+                    rids.Add(args.RuntimeValue);
                 }
-            }
 
-            if (generateBindingRedirects && context.TargetFramework.IsDesktop())
-            {
-                context.GenerateBindingRedirects(exporter, outputName);
```

Comment:
@davidfowl Why did you remove the binding redirect generation? It is causing one of the tests to fail. I will add it back.